pkg/gofr/datasource/pubsub/nats/message.go

```diff
@@ -0,0 +1,24 @@
+package nats
+
+import (
+	"github.com/nats-io/nats.go/jetstream"
+	"gofr.dev/pkg/gofr/datasource/pubsub"
+)
+
+type natsMessage struct {
+	msg    jetstream.Msg
+	logger pubsub.Logger
+}
+
+func newNATSMessage(msg jetstream.Msg, logger pubsub.Logger) *natsMessage {
+	return &natsMessage{
+		msg:    msg,
+		logger: logger,
+	}
+}
+
+func (nmsg *natsMessage) Commit() {
+	if err := nmsg.msg.Ack(); err != nil {
+		nmsg.logger.Errorf("unable to acknowledge message on Client JetStream: %v", err)
+	}
+}
```

Comment:
I would expect a commit to return an error

Reply:
Sorry but again the Commit() message is syntaxed like this to satisfy the `Committer` interface:

```go
type Committer interface {
	Commit()
}
```

We will address it too in a new issue along with Connect returning error

Reply:
Sure